include/nbl/builtin/hlsl/subgroup2/arithmetic_portability_impl.hlsl

```diff
@@ -0,0 +1,144 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_SUBGROUP2_ARITHMETIC_PORTABILITY_IMPL_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SUBGROUP2_ARITHMETIC_PORTABILITY_IMPL_INCLUDED_
+
+#include "nbl/builtin/hlsl/subgroup/arithmetic_portability_impl.hlsl"
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace subgroup2
+{
+
+namespace impl
+{
+
+template<class Binop, typename T, uint32_t ItemsPerInvocation, bool native>
+struct inclusive_scan
+{
+    using type_t = T;
+    using scalar_t = typename Binop::type_t;
```

Comment:
there's no need for `T` because Binop already contains it, `type_t` and `scalar_t` are the same thing, remove one of them

Reply:
comment valid for all 3 implementations

Reply:
aah sorry, the binop is on a scalar, the `T` is the vector we do it on.

In either case you have redundant template parameters, you can obtain `type_t = vector<scalar_t,ItemsPerInvocation>`